src/main/app/claims/models/interestDate.ts

```diff
@@ -4,9 +4,7 @@ import { MomentFactory } from 'common/momentFactory'
 
 export class InterestDate implements Serializable<InterestDate> {
 
-  type: string
-  date?: Moment
-  reason?: string
+  constructor (public type?: string, public date?: Moment, public reason?: string) {}
```

Comment:
isn't type mandatory anymore?

Reply:
To reflect real life scenario it should be. In code it cannot be because we use this `new X().deserialise({})` pattern which makes us have default constructors. The constructor behaviour is not changes, default constructor existed already.